deeplabcut/pose_estimation_tensorflow/predict_videos.py

```diff
@@ -868,15 +885,20 @@ def GetPoseF_GTF(cfg, dlc_cfg, sess, inputs, outputs, cap, nframes, batchsize):
     while cap.isOpened():
         if counter != 0 and counter % step == 0:
             pbar.update(step)
-        ret, frame = cap.read()
+        ret, _ = cap.read(frame_bgr)
         if ret:
-            frame = cv2.cvtColor(frame, cv2.COLOR_BGR2RGB)
             if cfg["cropping"]:
-                frames[batch_ind] = img_as_ubyte(
-                    frame[cfg["y1"] : cfg["y2"], cfg["x1"] : cfg["x2"]]
-                )
+                # If we need cropping, get a view of the bgr frame
+                frame_cropped_bgr = frame_bgr[cfg["y1"]: cfg["y2"], cfg["x1"]: cfg["x2"], :]
             else:
-                frames[batch_ind] = img_as_ubyte(frame)
+                # Otherwise, just point to the original bgr frame
+                frame_cropped_bgr = frame_bgr
+
+            if frame_cropped_bgr.dtype != np.uint8:
+                cv2.cvtColor(frame_cropped_bgr, cv2.COLOR_BGR2RGB, dst=frame_rgb)
```

Comment:
>  What do you think about getting rid of cv2.cvtColor and frame_rgb altogether, and simply use frame_bgr[..., ::-1]?

To be honest, many places don't deal well with slices, they well with "strided" data.

Here, I'm willing to bet that if you cropped the frame, that a contiguous copy of `fram_cropped_bgr` will be made, creating a short lived temporary input for `cv2.cvtColor`.

That said, tensorflow is quite fast, so who knows.

Do you have a benchmark we can run? That would be the best way.

Reply:
https://deeplabcut.github.io/DLC-inferencespeed-benchmark/

Reply:
https://github.com/DeepLabCut/DLC-inferencespeed-benchmark

Reply:
Thanks. I can try to find how to run these and compare before and after.